### PR TITLE
[DOCS] Remote cluster troubleshooting guide

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Adds a guide for troubleshooting remote clusters.

@ywangd I tried to make it into a generic remote cluster troubleshooting guide. Some topics are also applicable to the certificate based security model, like using the remote cluster info API to check if a connection was successful.

### CODE REVIEW

**Review by ywangd — APPROVED** [review]
LGTM

All the comments are minor and most of them are just updating the logging messages to be less "localish".

> I tried to make it into a generic remote cluster troubleshooting guide. Some topics are also applicable to the certificate based security model, like using the remote cluster info API to check if a connection was successful.

The RemoteClusterInfo API is a good point. We should suggest to always begin troubleshooting with this API to determine whether the desired security model is in use. That is, if API key based model is desired, the response must contain the `cluster_credentials` field and vice versa. This is the case even if everything else is mis-configured. First decide whether the local cluster is using the desired security model before proceeding to further troubleshooting.

That said, other than the RemoteClusterInfo API, most contents in the troubleshooting guide are specific to the API key based model. I think we should make this point clear. Thanks!

**Comment by abdonpijpelink** [comment]
> That said, other than the RemoteClusterInfo API, most contents in the troubleshooting guide are specific to the API key based model. I think we should make this point clear.

In https://github.com/elastic/elasticsearch/pull/99128/commits/2ba5e90f1be8059a95b495848aceb2ab6457531b, I've added a header "API key authentication issues". I've put "Remote cluster not reachable" and "TLS trust not established" under "Common issues", and the rest of the issues under "API key authentication issues".

**Comment by ywangd** [comment]
> I've added a header "API key authentication issues". I've put "Remote cluster not reachable" and "TLS trust not established" under "Common issues", and the rest of the issues under "API key authentication issues".

Sounds good to me. The sample messages for "TLS trust not established" are from the API key based remote cluster setup. But the general symptom is similar enough to apply to both. Thanks!

File: docs/reference/modules/cluster/remote-clusters-troubleshooting.asciidoc [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+cluster.
+
+====== Symptom
+
+The local cluster logs `failed to establish trust with server`:
+
+[source,txt,subs=+quotes]
+----
+[2023-06-29T09:40:55,465][WARN ][o.e.c.s.DiagnosticTrustManager] [local-node] *failed to establish trust with server* at [127.0.0.1]; the server provided a certificate with subject name [CN=remote_cluster], fingerprint [529de35e15666ffaa26afa50876a2a48119db03a], no keyUsage and no extendedKeyUsage; the certificate is valid between [2023-01-29T12:08:37Z] and [2032-08-29T12:08:37Z] (current time is [2023-08-16T23:40:55.464275Z], certificate dates are valid); the session uses cipher suite [TLS_AES_256_GCM_SHA384] and protocol [TLSv1.3]; the certificate has subject alternative names [DNS:localhost,DNS:localhost6.localdomain6,IP:127.0.0.1,IP:0:0:0:0:0:0:0:1,DNS:localhost4,DNS:localhost6,DNS:localhost.localdomain,DNS:localhost4.localdomain4,IP:192.168.0.42]; the certificate is issued by [CN=Elastic Auto RemoteCluster CA] but the server did not provide a copy of the issuing certificate in the certificate chain; this ssl context ([(shared) (with trust configuration: JDK-trusted-certs)]) is not configured to trust that issuer but trusts [97] other issuers
```
Comment (ywangd) [inline docs/reference/modules/cluster/remote-clusters-troubleshooting.asciidoc:93]:
```suggestion
[2023-06-29T09:40:55,465][WARN ][o.e.c.s.DiagnosticTrustManager] [local-node] *failed to establish trust with server* at [192.168.0.42]; the server provided a certificate with subject name [CN=remote_cluster], fingerprint [529de35e15666ffaa26afa50876a2a48119db03a], no keyUsage and no extendedKeyUsage; the certificate is valid between [2023-01-29T12:08:37Z] and [2032-08-29T12:08:37Z] (current time is [2023-08-16T23:40:55.464275Z], certificate dates are valid); the session uses cipher suite [TLS_AES_256_GCM_SHA384] and protocol [TLSv1.3]; the certificate has subject alternative names [DNS:localhost,DNS:localhost6.localdomain6,IP:127.0.0.1,IP:0:0:0:0:0:0:0:1,DNS:localhost4,DNS:localhost6,DNS:localhost.localdomain,DNS:localhost4.localdomain4,IP:192.168.0.42]; the certificate is issued by [CN=Elastic Auto RemoteCluster CA] but the server did not provide a copy of the issuing certificate in the certificate chain; this ssl context ([(shared) (with trust configuration: JDK-trusted-certs)]) is not configured to trust that issuer but trusts [97] other issuers
```

Missed a 127.0.0.1 entry in the previous review

File: docs/reference/modules/cluster/remote-clusters-troubleshooting.asciidoc [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+* Is hostname verification failing? 
+* Is the certificate expired?
+
+Once you know the cause, you should be able to fix it by adjusting the remote
+cluster related SSL settings on either the local cluster or the remote cluster. 
+
+Often, the issue is on the local cluster. Fix it by either configuring necessary
+trusted CAs (`xpack.security.remote_cluster_client.ssl.certificate_authorities`)
+or adjusting the verification mode
+(`xpack.security.remote_cluster_client.ssl.verification_mode`).
```
Comment (ywangd) [inline docs/reference/modules/cluster/remote-clusters-troubleshooting.asciidoc:120]:
```suggestion
Often, the issue is on the local cluster. For example, fix it by configuring necessary
trusted CAs (`xpack.security.remote_cluster_client.ssl.certificate_authorities`).
```

Adjusting verificaton mode seems to suggest disabling hostname verification which is not what we would recommend. It is a risk for server side only TLS.

**Review by ywangd — APPROVED** [review]

**Pull request merged**